org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/contentmodel/XMLSchemaHoverExtensionsTest.java

```diff
@@ -304,6 +304,11 @@ public void hoverTextWithUnion3() throws BadLocationException, MalformedURIExcep
 						"Source: [dressSize.xsd](" + schemaURI + ")",
 				r(2, 52, 3, 9));
 	}
+	
+	@Test
+	public void hoverText() throws BadLocationException {

```

Comment:
This test passes even if the null check isn't present.

Reply:
It's because exception is cached, so we don't see the problem. But this PR clean the usecase and doesn't throw the NPE exception.

Reply:
Okay. I don't see the point of including this test case if it doesn't reproduce the issue.

Reply:
I would like to keep it because without this fix we see a error stack trace when test is executed.

Reply:
Okay. I can see why this would be helpful, but its very easy to miss this exception in the output. I think it would be better to write a unit test on `ContentModelHoverParticipant.onText()` so that the build will fail if this behaviour is changed.

Reply:
I think a test like this would be more helpful:

```java
@Test
public void hoverWithNullParentNode() throws Exception {
	ContentModelHoverParticipant hoverParticipant = new ContentModelHoverParticipant();
	IHoverRequest hoverRequest = new IHoverRequest() {
		// default auto-suggested implementation for other methods
		@Override
		public DOMNode getNode() {
			return new DOMText(0, 0);
		}
			
	};
	assertNull(hoverParticipant.onText(hoverRequest));
}
```

Reply:
See #1001